Allow usage with ember-cli-fastboot.

rwjblue opened the pull request:
* `renderer:-dom` is provided in versions of Ember > 1.10.
* Actually test the real document.title in `ember test` (to help ensure this is really working).

kimroen commented:
Interesting, I wonder if we could test that it works with fastboot.

:+1: on the changes!

rwjblue commented:
> I wonder if we could test that it works with fastboot.

A few more changes are required in Fastboot itself to allow this, but I am working on those next. The first step was to remove the usage of `window.document` in that scenario (because `window` or `document` are undefined in Node-land).


@kimroen  - Working on another PR to confirm that it works on all Ember versions from 1.10 and up.

kimroen commented:
@rwjblue Ah, using `ember-try`? Thanks.

rwjblue commented:
@kimroen - confirm

rwjblue commented:
OK, this should be ready to go.

* Added ember-try testing 1.10, 1.11, release, beta, and canary.
* Added unit test for `Router.prototype.setTitle`.
* Ensured that the `document.title` is cleaned up after each test.

Travis will now run one build per Ember version, so that a failure in a specific version can be spotted easily. This is in line with the current ember-cli blueprint for new addons (unreleased).

See the travis build here: https://travis-ci.org/kimroen/ember-cli-document-title/builds/59563596

kimroen commented on tests/acceptance/document-title-test.js:
```diff
@@ -2,16 +2,17 @@ import Ember from 'ember';
 
   afterEach: function() {
     Ember.run(application, 'destroy');
+    document.title = document.title;
```
Should this be

```js
document.title = originalTitle;
```

rwjblue replied:
heheh, yes

rwjblue replied:
Updated. Good catch.

kimroen commented:
Cool - this is pretty rad. Thanks again.

I'll merge and release this in a little bit.

kimroen commented:
Just as a note: I'm not sure I understand what `_dom.document.title` does - will it set the `<title>` tag that is in the served index.html?

rwjblue commented:
It doesn't do anything.... Yet

danmcclain commented:
@kimroen Yes, in client side ember, it ends up overwriting the title, but on the server (fastboot) side, it doesn't do anything, like @rwjblue just said

rwjblue commented:
Well, that wasn't exactly correct. 

In a browser (as of 1.11 or 1.12) '_dom.document' is just 'window.document'. 

In node it does not blow up (which is an improvement), but it doesn't cause the fastboot server to add the title to < head> yet. I'm still working on that part.

kimroen commented:
Okay, so it doesn't really "add support for fastboot" then, but it will likely Just Work™ once that's taken care. Just trying to make sure I don't lie in the changelog :)

kimroen commented:
Thanks again - available as version `0.1.0`.

rwjblue commented:
@kimroen - Correct it does not add support, it allows this addon to be used if you are using Fastboot.  Without these tweaks, having this addon in an application that wanted to use fastboot would throw an error (because `window` and `document` are undefined).

kimroen commented:
@rwjblue Understood!

On Wed, Apr 22, 2015 at 8:09 PM, Robert Jackson <notifications@github.com>
wrote:

> @kimroen - Correct it does not add support, it allows this addon to be used if you are using Fastboot.  Without these tweaks, having this addon in an application that wanted to use fastboot would throw an error (because `window` and `document` are undefined).  
> ---
> Reply to this email directly or view it on GitHub:
> https://github.com/kimroen/ember-cli-document-title/pull/13#issuecomment-95287710

rwjblue commented:
A little followup here: using https://github.com/tildeio/ember-cli-fastboot/pull/22 along with 0.1.0 of ember-cli-document-title works properly when rendered via ember-cli-fastboot.

kimroen commented:
Thanks for the update - very exciting!
